src/report/entry.cpp

```diff
@@ -25,21 +25,6 @@ namespace report {
     return entry;
 }
 
-Entry &Entry::operator=(Entry value) {
-    nlohmann::json::operator=(value);
-    return *this;
-}
-
-Entry &Entry::operator[](std::string key) {
-    // Note: out_of_range could not happen because operator[] returns
-    // a null json object if the accessed key is missing
-    try {
-        return static_cast<Entry &>(nlohmann::json::operator[](key));
-    } catch (std::domain_error &) {
-        throw JsonDomainError();
-    }
-}
-
```

Comment:
The reason for doing this is that now that we made `nlohmann::json` public it's no longer necessary to manually expose the methods that we need to support. Having these two methods in place lead to, in particular, the issue that it was not possible to access keys that were not of string type (for example to access the index of an array).